be/src/formats/parquet/column_reader.cpp

```diff
@@ -288,6 +321,92 @@ bool ScalarColumnReader::_column_all_pages_dict_encoded() {
     return true;
 }
 
+void ScalarColumnReader::collect_column_io_range(std::vector<io::SharedBufferedInputStream::IORange>* ranges,
+                                                 int64_t* end_offset, ColumnIOType type, bool active) {
+    const auto& column = _opts.row_group_meta->columns[_field->physical_column_index];
+    if (type == ColumnIOType::PAGES) {
+        const tparquet::ColumnMetaData& column_metadata = column.meta_data;
+        if (_offset_index_ctx != nullptr) {
+            // add dict page
+            if (column_metadata.__isset.dictionary_page_offset) {
+                auto r = io::SharedBufferedInputStream::IORange{
+                        .offset = column_metadata.dictionary_page_offset,
+                        .size = column_metadata.data_page_offset - column_metadata.dictionary_page_offset,
+                        .active = active};
+                ranges->emplace_back(r);
+                *end_offset = std::max(*end_offset, r.offset + r.size);
+            }
+            _offset_index_ctx->collect_io_range(ranges, end_offset, active);
+        } else {
+            int64_t offset = 0;
+            if (column_metadata.__isset.dictionary_page_offset) {
+                offset = column_metadata.dictionary_page_offset;
+            } else {
+                offset = column_metadata.data_page_offset;
+            }
+            int64_t size = column_metadata.total_compressed_size;
+            auto r = io::SharedBufferedInputStream::IORange{.offset = offset, .size = size, .active = active};
+            ranges->emplace_back(r);
+            *end_offset = std::max(*end_offset, offset + size);
+        }
+    } else if (type == ColumnIOType::PAGE_INDEX) {
+        // only active column need column index
```

Comment:
so page index data will be taken as a separated io range?

Reply:
Yes, all pageindex will be saved together that is next to the footer.